SpinCore_pp/save_data.py

```diff
@@ -0,0 +1,74 @@
+import h5py
+import os
+import pyspecdata as psd
+import pyspecProcScripts
+import subprocess
+
+
+def save_data(dataset, target_directory, config_dict, counter_type):
+    """save data to an h5 file with appropriately labeled nodename and performs
+    rough processing
+
+    Parameters
+    ==========
+    dataset: nddata
+        acquired data in nddata format
+    target_directory: str
+        directory on the share drive you want to save to
+    config_dict: dict
+        config_dict pulled from the active.ini file
+    counter_type: str
+        type of counter you are incrementing
+
+    Returns
+    =======
+    config_dict: dict
+        the updated config dict after appropriately incrementing the counter
+    """
+    # {{{ create filename
+    filename_out = f"{config_dict['date']}_{config_dict['chemical']}_{config_dict['type']}"+".h5"
+    # }}}
+    nodename = config_dict["type"] + "_" + str(config_dict["%s_counter" % counter_type])
+    dataset.name(nodename)
+    if os.path.exists(f"{target_directory}{filename_out}"):
+        print("this file already exists so we will add a node to it!")
+        with h5py.File(
+            os.path.normpath(os.path.join(target_directory, f"{filename_out}"))
+        ) as fp:
+            while nodename in fp.keys():
+                config_dict["%s_counter" % counter_type] += 1
+                nodename = (
+                    config_dict["type"]
+                    + "_"
+                    + str(config_dict["%s_counter" % counter_type])
+                )
+            dataset.name(nodename)
+    dataset.hdf5_write(f"{filename_out}", directory=target_directory)
+    print("\n** FILE SAVED IN TARGET DIRECTORY ***\n")
+    my_exp_type = '/'.join([os.path.split(os.path.split(os.path.split(target_directory)[0])[0])[1], os.path.split(os.path.split(target_directory)[0])[1]]) 
```

Comment:
takes the last two directories so this prints ODNP_NMR_comp/Echoes for example if the target directory = 'C:\\Users\\aguinnes\\Syracuse University\\Group-FranckGroup - General\\exp_data\\ODNP_NMR_comp\\Echoes\\' try it in ipython it works

Reply:
objectively, what is there is not good.

Reply:
so are we going back to feeding the function my_exp_type instead of target_directory?